### PR TITLE
支持 Near

### PR DESCRIPTION
**Pull request by huazhouwang**
## What does this implement/fix? Explain your changes.
1. 调整 BaseRestfulClient 命名为 SimpleClient
2. 支持 NEAR
3. 测试 NEAR

## Does this close any currently open issues?

If it fixes a bug or resolves a feature request, be sure to link to that issue.
…

## Pull request type

_Put an `x` in the boxes that apply_

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Where has this been tested?

…

## Any other comments?

…

### CODE REVIEW

**Review by taimanhui — APPROVED** [review]

File: src/provider/chains/near/nearcli.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    return {
+      bestBlockNumber: blockNumber,
+      isReady,
+    };
+  }
+
+  async getBestBlock(): Promise<{ blockNumber: number; blockHash: string }> {
+    const resp: any = await this.rpc.call('status', []);
+    return {
+      blockNumber: Number(resp.sync_info.latest_block_height),
```
[the remark this answers is not in the document]
Comment (huazhouwang) [reply src/provider/chains/near/nearcli.ts:59]: 用 Number 好点吧，它会自动处理十进制和十六进制

File: src/provider/chains/near/nearcli.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        price: normalPrice,
+      },
+    };
+  }
+
+  async getTransactionStatus(txid: string): Promise<TransactionStatus> {
+    try {
+      const resp: any = await this.rpc.call('tx', [
+        txid,
+        'dontcare', // required tx signer actually, but I found that just filling in any string works fine
```
[the remark this answers is not in the document]
Comment (huazhouwang) [reply src/provider/chains/near/nearcli.ts:166]: 嗯，文档是这么写的

File: src/provider/chains/near/provider.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import { BaseProvider } from '../../abc';
+
+import { NearCli } from './nearcli';
+import { PublicKey } from './sdk/key_pair';
+import * as nearTx from './sdk/transaction';
+
+const FT_TRANSFER_GAS = '30000000000000';
+const FT_TRANSFER_DEPOSIT = '1';
+
+const IMPLICIT_ACCOUNT_PATTERN = /^[a-z\d]{64}$/;
```
Comment (somebodyLi) [inline src/provider/chains/near/provider.ts:18]: 需要用正则区分类型？

**Review by somebodyLi — APPROVED** [review]

File: src/provider/chains/near/provider.ts (hunk 3 above)
Comment (huazhouwang) [reply src/provider/chains/near/provider.ts:18]: 嗯，因为 ImplicitAccount 是特殊的 AccountID

**Pull request merged**